tutorials/llm/slimpajama/data_pipeline.ipynb

```diff
@@ -0,0 +1,327 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "concat_task = run.Script(\"/nemo_run/code/data/concat.sh\", args=[\"/data/slimpajama/train\", \"1\"])\n",
+    "concat_task"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preprocess Data\n",
+    "\n",
+    "This final step preprocesses the JSONL files to the BIN and IDX files required by NeMo and Megatron Core. It uses the `preprocess_data` function defined [here](./data/preprocess.py)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "preprocess_data??"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "preprocess_task = run.Partial(\n",
+    "    preprocess_data,\n",
+    "    data_dir=\"/data/slimpajama\",\n",
+    "    output_dir=\"/data/slimpajama_megatron\",\n",
+    "    tokenizer_model=\"/data/tokenizer/tokenizer.model\",\n",
+    "    tokenizer_library=\"sentencepiece\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "preprocess_task"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Put it all together\n",
+    "\n",
+    "Now that all the tasks are configured, lets define an executor to run them on and an experiment to run them sequeuntially. \n",
+    "\n",
+    "> [!NOTE]\n",
+    "> Each task can be run individually or in any combination. The notebook runs all tasks sequentially. To remove a task, just remove the corresponding `exp.add(...)` for that corresponding task.\n",
+    "> This customization is handy if you already have JSONL files processed, for example, from NeMo-Curator."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's define a local executor to run the experiment locally.\n",
+    "def docker_executor(host_data_path: str):\n",
+    "    packager = run.GitArchivePackager(subpath=\"examples/llm/slimpajama\") # This will package all code inside the folder. NOTE: only committed changes are packaged, so if you make a change, make sure to commit it.\n",
+    "    executor = run.DockerExecutor(\n",
+    "        packager=packager,\n",
+    "        ipc_mode=\"host\",\n",
+    "        shm_size=\"30g\",\n",
+    "        env_vars={\"PYTHONUNBUFFERED\": \"1\"},\n",
+    "        volumes=[f\"{host_data_path}:/data\"],\n",
+    "        container_image=\"python:3.11\",\n",
+    "        ulimits=[\"memlock:-1\", \"stack:67108864\"],\n",
+    "    )\n",
+    "    return executor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "executor = docker_executor(host_data_path=\"/home/hemild/dev/data\")\n",
```

Comment:
Best to remove PII like host_data_path here.

Keep this on a separate line with instructions to the user to configure this.